Meet unable to extract decompressor error when building on macos

**Operating system**

Macos

**Cmake version**
cmake version 3.23.3

**Ninja version**
1.11.0

**Compiler name and version**

**Full cmake and/or ninja output**
[18/18] cd /Users/jianzjzhang/Documents/ClickHouse/build/programs/self-extracting && /usr/local/Cellar/cmake/3.23.3/b...&& /Users/jianzjzhang/Documents/ClickHouse/build/utils/self-extracting-executable/compressor clickhouse ../clickhouse
FAILED: programs/self-extracting/CMakeFiles/self-extracting /Users/jianzjzhang/Documents/ClickHouse/build/programs/self-extracting/CMakeFiles/self-extracting 
cd /Users/jianzjzhang/Documents/ClickHouse/build/programs/self-extracting && /usr/local/Cellar/cmake/3.23.3/bin/cmake -E remove clickhouse && /Users/jianzjzhang/Documents/ClickHouse/build/utils/self-extracting-executable/compressor clickhouse ../clickhouse
Error: unable to extract decompressor
ninja: build stopped: subcommand failed.

It seems that "set (DECOMPRESSOR "--decompressor=${CMAKE_BINARY_DIR}/utils/self-extracting-executable/decompressor")" should also be set in the else branch in "ClickHouse/programs/self-extracting/CMakeLists.txt"

--- a/utils/self-extracting-executable/post_build.sh
+++ b/utils/self-extracting-executable/post_build.sh
@@ -1,3 +1,7 @@
 padding="               "
-sz="$(stat -c %s 'decompressor')"
+if [[ $OSTYPE == 'darwin'* ]]; then
+    sz="$(stat -f %z 'decompressor')"
+else
+    sz="$(stat -c %s 'decompressor')"
+fi
 printf "%s%s" "${padding:${#sz}}" $sz

